### PR TITLE
drivers:platform:pico:pico_i2c: Add dynamic config for baud rate

### PR DESCRIPTION
**Pull request by rbolboac**
Add dynamic configuration for baud-rate to be able to support multiple I2C devices on the same I2C hardware instance.

### CODE REVIEW

File: drivers/platform/pico/pico_i2c.c [hunk 1]
```diff
@@ -74,7 +96,7 @@ int32_t pico_i2c_init(struct no_os_i2c_desc **desc,
 	pico_i2c = (struct pico_i2c_desc *)calloc(1,sizeof(*pico_i2c));
 	if (!pico_i2c) {
 		ret = -ENOMEM;
-		goto error;
+		goto free_desc;
```
Comment (amiclaus) [inline drivers/platform/pico/pico_i2c.c:99]: I believed this change is unrelated to the commit subject.

File: drivers/platform/pico/pico_i2c.c [hunk 2]
```diff
@@ -106,6 +128,7 @@ int32_t pico_i2c_init(struct no_os_i2c_desc **desc,
 	return 0;
 error:
 	free(pico_i2c);
+free_desc:
```
Comment (amiclaus) [inline drivers/platform/pico/pico_i2c.c:131]: ditto.

File: drivers/platform/pico/pico_i2c.c (hunk 1 above)
Comment (rbolboac) [reply drivers/platform/pico/pico_i2c.c:99]: Thank you! I added another commit with these changes.

**Review by amiclaus — APPROVED** [review]

**Comment by rbolboac** [comment]
- removed typo in commit message

**Review by buha — APPROVED** [review]
